lib/SILOptimizer/Mandatory/TFDevicePartition.cpp

```diff
@@ -270,7 +270,11 @@ class DevicePartitionCloner
 }  // end anonymous namespace
 
 void DevicePartitionCloner::visitGraphOperationInst(GraphOperationInst *inst) {
-  if (!targetOps.count(inst)) return;
+  // TODO: try and remove this special case.
+  if (inst->getName().str() == "tf_tensor_to_i1") {
```

Comment:
Use ‘.equals()’ for efficiency.

Reply:
Will fix in the prior PR https://github.com/apple/swift/pull/17698

Reply:
Yes, but when you call str(), it creates a std::string first instead of directly comparing.

Reply:
Note `getName()` returns an `Identifier`, so `str()` returns a StringRef, not std::string: https://github.com/apple/swift/blob/ddaec4711af36b8b60c532b869b73b9ad947580b/include/swift/AST/Identifier.h#L65

Reply:
Ok. I thought `getName()` returned a `StringRef`. Thanks for clarifying.